sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -305,10 +325,12 @@ bool CurlSession::isUploadRequest()
 }
 
 // Send buffer thru the wire
-CURLcode CurlSession::SendBuffer(uint8_t const* buffer, size_t bufferSize)
+CURLcode CurlSession::SendBuffer(uint8_t const* buffer, size_t bufferSize, Context const& context)
 {
   for (size_t sentBytesTotal = 0; sentBytesTotal < bufferSize;)
   {
+    // check cancelation for each chunk of data
+    context.ThrowIfCanceled();
```

Comment:
Why not in the nested for-loop?

Reply:
It would be the same thing,
The inner loop is expected to run no more than 2 iterations at the worst case.
And, on that worst case, the first iteration will call `waitForSocket` which will check the cancellation token again. So, inner loop is checked by the `waitForSocket` call

Reply:
Where's the call to `waitForSocket` in the nested loop? Also, I am a bit confused by this loop condition:
`for (CURLcode sendResult = CURLE_AGAIN; sendResult == CURLE_AGAIN;)`, especially since it looks like `sendResult` can continuously be `CURLE_AGAIN`. Please add some comments which explain why this loop won't iterate more than twice.

Reply:
Updated comments.
I said `waitForSocket` but it is now called `pollSocketUntilEventOrTimeout` 

it is still waiting for a socket to be ready to be used,